src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -207,53 +438,95 @@ t8_forest_write_vtk_via_API (t8_forest_t forest, const char *fileprefix,
     t8_locidx_t         offset =
       t8_forest_get_tree_element_offset (forest, itree);
     /* We iterate over all elements in the tree */
+    /* Compute the global tree id */
+    gtreeid = t8_forest_global_tree_id (forest, itree);
     for (ielement = 0; ielement < elems_in_tree; ielement++) {
-
       t8_element_t       *element =
         t8_forest_get_element_in_tree (forest, itree, ielement);
       T8_ASSERT (element != NULL);
       vtkSmartPointer < vtkCell > pvtkCell = NULL;
       t8_element_shape_t  element_shape = scheme->t8_element_shape (element);
-      int                 num_corners =
-        scheme->t8_element_num_corners (element);
-
+      num_corners = t8_get_number_of_vtk_nodes (element_shape, curved_flag);
```

Comment:
The variable name corners is misleading, since the nodes do not lie on the corners necessarily